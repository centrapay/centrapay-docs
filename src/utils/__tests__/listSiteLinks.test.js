const { listSiteLinks } = require('../listSiteLinks');

describe('listSiteLinks', () => {
  it('should return a unique set of links and anchor tags', () => {
    expect(listSiteLinks('src/utils/__tests__/examples/duplicate-paths')).toMatchSnapshot();
  });

  it('should not include any external links', () => {
    expect(listSiteLinks('src/utils/__tests__/examples/external-links')).toMatchSnapshot();
  });

  it('should not include anchor tags that do not go anywhere', () => {
    expect(listSiteLinks('src/utils/__tests__/examples/empty-anchor')).toMatchSnapshot();
  });
});
