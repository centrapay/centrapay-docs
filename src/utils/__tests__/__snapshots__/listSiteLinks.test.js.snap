// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`listSiteLinks > should return the set of possible site links 1`] = `
Set {
  "/utils/__tests__/examples/happy/api",
  "/utils/__tests__/examples/happy/api#heading-1",
  "/utils/__tests__/examples/happy/api#heading-2",
  "/utils/__tests__/examples/happy/api#heading-3",
  "/utils/__tests__/examples/happy/api#heading-4",
  "/utils/__tests__/examples/happy/api#heading-5",
  "/utils/__tests__/examples/happy/api#heading-6",
  "/utils/__tests__/examples/happy/api#heading-7",
  "/utils/__tests__/examples/happy/api#included-only-once",
  "/utils/__tests__/examples/happy/api/payments",
  "/utils/__tests__/examples/happy/api/payments#payments-heading",
  "/utils/__tests__/examples/happy/guides",
  "/utils/__tests__/examples/happy/guides#guides-heading",
}
`;
